test/csuite/parse_opts/main.c

```diff
@@ -0,0 +1,315 @@
+ * using testutil_parse_opt to parse any we don't recognize, those are put into TEST_OPTS.
+ */
+static const char *command_line0[] = {"parse_opts", "-b", "builddir", "-T", "21", NULL};
+static const char *command_line1[] = {"parse_opts", "-bbuilddir", "-T", "21", NULL};
+static const char *command_line2[] = {"parse_opts", "-v", "-PT", NULL};
+static const char *command_line3[] = {"parse_opts", "-v", "-Po", "my_store", "-PT", NULL};
+static const char *command_line4[] = {"parse_opts", "-v", "-Pomy_store", "-PT", NULL};
+
+static const char *command_line5[] = {"parse_opt", "-vd", "-Pomy_store",
+  "-c"
+  "string_opt",
+  "-PT", NULL};
+static const char *command_line6[] = {
+  "parse_opt", "-dv", "-Pomy_store", "-cstring_opt", "-PT", NULL};
+static const char *command_line7[] = {
+  "parse_opt", "-ev", "-cstring_opt", "-Pomy_store", "-PT", "-f", "22", NULL};
+static const char *command_line8[] = {"parse_opt", "-evd", "-Pomy_store", "-PT", "-f22", NULL};
+static const char *command_line9[] = {"parse_opt", "-v", "-Pomy_store", "-PT", NULL};
+
+static const char **command_lines[] = {
+  command_line0,
+  command_line1,
+  command_line2,
+  command_line3,
+  command_line4,
+  command_line5,
+  command_line6,
+  command_line7,
+  command_line8,
+  command_line9,
+};
+
+static TEST_OPTS expected[] = {
+  {NULL, NULL, {0}, NULL, (char *)"builddir", NULL, 0, NULL, NULL, false, false, false, false, 0, 0,
+    21, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
+  {NULL, NULL, {0}, NULL, (char *)"builddir", NULL, 0, NULL, NULL, false, false, false, false, 0, 0,
+    21, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
+  /* If -PT is used, the tiered_storage source is set to dir_store, even if -Po is not used. */
+  {NULL, NULL, {0}, NULL, NULL, (char *)"dir_store", 0, NULL, NULL, false, false, true, true, 0, 0,
+    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
+  {NULL, NULL, {0}, NULL, NULL, (char *)"my_store", 0, NULL, NULL, false, false, true, true, 0, 0,
+    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
+  {NULL, NULL, {0}, NULL, NULL, (char *)"my_store", 0, NULL, NULL, false, false, true, true, 0, 0,
+    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
+
+  /* Extended tests */
+  {NULL, NULL, {0}, NULL, NULL, (char *)"my_store", 0, NULL, NULL, false, false, true, true, 0, 0,
+    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
+  {NULL, NULL, {0}, NULL, NULL, (char *)"my_store", 0, NULL, NULL, false, false, true, true, 0, 0,
+    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
+  {NULL, NULL, {0}, NULL, NULL, (char *)"my_store", 0, NULL, NULL, false, false, true, true, 0, 0,
+    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
+  {NULL, NULL, {0}, NULL, NULL, (char *)"my_store", 0, NULL, NULL, false, false, true, true, 0, 0,
+    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
+  {NULL, NULL, {0}, NULL, NULL, (char *)"my_store", 0, NULL, NULL, false, false, true, true, 0, 0,
+    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
+};
+
+static EXTENDED_OPTS x_expected[] = {
+  {NULL, 0, 0, 0},
```

Comment:
Similar comment here to help someone understand the relation between the command and an extended structure initialization.